### ISSUE TITLE
Inf-scroll: doClaimSearch() sometimes returns duplicates, causing scrolling problems.

### ISSUE THREAD

**Issue by infinite-persistence** [lbryio/lbry-desktop#4408]
When the requested `page` and returned `results` don't match, we'll get the `Encountered two children with the same key` error. #4403 tried to solve it by making sure these 2 variables are "in sync", but it seems like repeated identical queries can result in the same error as well.  I'm not familiar with the redux stuff yet to dig further.

Here's a video to explains the issue:
https://youtu.be/K4iMKJu_OtM

**Comment by tzarebczan** [comment]
Pretty sure this happens when claim search results change due to new trending / new content between the original search and when you scroll down. The solution would be to ignore any keys already loaded and don't duplicate the content.

**Comment by infinite-persistence** [comment]
Yeah, I wondered if that is the solution as well.  But given that it happens without scrolling or when scrolling has long stopped (as shown in the video), I wonder if it's a mistake in the redux side.  
It also happened on obscure tag searches (limited & same results for several days), so it can't be due to actual new entries in between those searches.

**Comment by infinite-persistence** [comment]
#### Root-cause
Currently, the code calls `doClaimSearch()` for the same query if the system clock has incremented to the next minute.  This seems intended from the following code:
```
// Don't use the query from createNormalizedClaimSearchKey for the effect since that doesn't include page & release_time
const optionsStringForEffect = JSON.stringify(options);
```
Over at `doClaimSearch()`, the `append` is set to true if we have scrolled, and ...
```
append: options.page && options.page !== 1,
```
... duplicate check hasn't been implemented.
```
  if (append) {
    // todo: check for duplicate urls when concatenating?
    claimSearchByQuery[query] = claimSearchByQuery[query] && claimSearchByQuery[query].length ? claimSearchByQuery[query].concat(urls) : urls;
  } else {
```
#### Possible fixes
1. **The client side should exclude `release_time` from the consideration to call `doClaimSearch()`, as it causes a redundant call.**
   _However, given the code comment above, it seems like we need to include `release_time`.  What scenario does it cover?_
2. **The redux side should exclude duplicates before appending.**
   _This will solve all future issues with duplication, but I think it will mask mistakes of the client._
_Also, constantly scanning the list (could be huge?) for duplicates will waste CPU cycles._

I'm leaning towards solution 1.  @seanyesmunt, your thoughts?

**Comment by seanyesmunt** [comment]
I think I would prefer fix #1 mostly because of this comment
>  I think it will mask mistakes of the client

Generally, I have started to prefer putting more logic inside of components. I think that makes the data flow easier to track. It does lead to component bloat though, but that is out of scope for this issue.

**Comment by infinite-persistence** [comment]
## tl;dr
After experimenting, I think neither solution will fix it completely;  might need more elaborate stuff at redux.  Live with the minor glitches for now, I guess.

## Details
#### Solution 1 will just mask the issue as well
I think it's actually due to the hack that prevents showing videos from the future:
```
  // Hack for at least the New page until https://github.com/lbryio/lbry-sdk/issues/2591 is fixed
  options.release_time = `<${Math.floor(
    moment()
      .startOf('minute')
      .unix()
  )}`;
```
Say that you are currently showing 3 pages (e.g. 20, 20, 20) ... and after 1 minute or more, the next page query is essentially a different one due to the new `release_time`, but we continue to append the 4th page (e.g. 20, 20, 20, **20**).  I'm guessing there may be new items now in the upper pages for this new search, so the list got pushed down and we see the duplicates in the final 20 that we take in.

So, I tried "locking" the `release_time` when getting new pages.  It now takes longer for the issue to happen, but it will still happen.  My guess is that new confirmations happened that now satisfies the `release_time`, so these new items are being injected in the next search result, and we are back to the same problem.

Next, I tried subtracting a few minutes to the locked time -- a 15 minute buffer gave me the longest duration without issues, but eventually the error still happened.

#### Solution 2 might not be perfect either
Simply removing the duplicates in the new page would result in a partially-filled page;  the current code takes partial pages as "no more results", so scrolling would end.  Seems like there's no quick-fix unless we add elaborate code to the current "append-only" method.
